client/src/pages/Map.js

```diff
@@ -0,0 +1,124 @@
+import { MapContainer, TileLayer, Marker, Popup } from "react-leaflet";
+import MarkerClusterGroup from "react-leaflet-markercluster";
+import CheckInText from "../components/CheckInText";
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+export default function Map() {
+  const [map, setMap] = useState(null);
+  const [playGroundData, setPlayGroundData] = useState([]);
+  const [state, setState] = useState();
+  const { params } = useParams([]);
+
+  useEffect(() => {
+    const url = "/api/playground";
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => {
+        setPlayGroundData(data);
+      })
+      .catch((error) => console.error(error));
+
+    const searchInputUrl = `https://nominatim.openstreetmap.org/search?q=${params}&limit=20&format=json`;
+    fetch(searchInputUrl)
+      .then((res) => res.json())
+      .then((data) => {
+        const newLatitude = Number(data[0]?.lat);
+        const newLongitude = Number(data[0]?.lon);
+        map.flyTo([newLatitude, newLongitude], 15);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }, [state, params, map]);
+
+  function handleOnSubmit(e) {
+    e.preventDefault();
+    const form = e.target;
+    const formInputValue = form.searchInput.value;
+    const searchInputUrl = `https://nominatim.openstreetmap.org/search?q=${formInputValue}&limit=20&format=json`;
+    fetch(searchInputUrl)
+      .then((res) => res.json())
+      .then((data) => {
+        const newLatitude = Number(data[0]?.lat);
+        const newLongitude = Number(data[0]?.lon);
+        map.flyTo([newLatitude, newLongitude], 15);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+    form.reset();
+  }
+
+  function handleCheckButton(data) {
+    const url = `api/playground/${data._id}`;
+    const patchMethodCheckIn = {
+      method: "PATCH",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        userId: JSON.parse(localStorage.getItem("userId")),
+      }),
+    };
+    fetch(url, patchMethodCheckIn)
+      .then((res) => {
+        setState(!state);
+        res.json();
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }
+
+  return (
+    <>
+      <form onSubmit={handleOnSubmit}>
+        <input
+          name="searchInput"
+          id="searchInput"
+          placeholder="PLZ oder Stadteil"
+        />
```

Comment:
A `label` would help blind people to know what to input. Right now for a screen reader this is just an input.